### ISSUE TITLE
Activity Streams and Unified Search?

### ISSUE THREAD

**Issue by jonlachlan** [MaksTr/meteor-mantra-kickstarter#45]
In order to make an ecosystem of general-use component modules, it seems likely that you'd soon need a way to connect to an activity stream that logs and notifies updates from a variety of modules. And you would want content to be indexed so that they are available in a unified search. 

Looking into activity streams on node, I came across activitystrea.ms API (https://www.npmjs.com/package/activitystrea.ms), so there's already a standard data structure for this. There is also Activity Streams App (https://www.npmjs.com/package/activitystreams), which uses Neo4J and Redis. 

Then there's unified search, a necessity in many use cases, so somehow the modules would need to define how to index fields for search. Is there away to attach onto collections, and should module authors be encouraged to specify search index definitions?

Both of these would require a bit more from a module author, so I'd suggest that both Activity Steam and Search Indexing it could be optional.

I have been considering the idea of a system with goals similar to Mantra's (and perhaps it will itself be a Mantra app), but I'm still writing out the proposed specs before I post it.

**Comment by makstr** [comment]
I agree, this is a common functionality and a must in many apps. We should all look into this. Here is a service I wanted to explore but never had a chance. Anyone has experience with getstream.io?

http://blog.getstream.io/post/136611889848/building-news-feeds-activity-streams-with-meteor
